decode.go

```diff
@@ -20,24 +38,33 @@ func readRune(r RuneReader) (c rune, size int, err error) {
 	} else if e != nil {
 		return c, s, e
 	}
-
 	// check to see if this is a valid emoji rune
-	_, exists := revEmojis[c]
-	if !exists && c != padding && c != padding40 && c != padding41 && c != padding42 && c != padding43 {
-		return 0, 0, errors.New("Invalid rune " + string(c))
+	if *ecojiV1 {
+		if ok := checkRuneV1(c); !ok {
+			return 0, 0, errors.New("Invalid rune " + string(c))
+		}
+	} else {
+		if ok := checkRune(c); !ok {
+			// try to fallback to ecoji v1
+			if isV1 := checkRuneV1(c); isV1 {
```

Comment:
This may need to be more strict for the following case.
 1. See a rune that is only used by ecoji v2
 2. See a run that is only used by ecoji v1

For the above sequence of events, probably want to return an error.  I don't think it currently does, but not completely sure.   

In other programming languages I might use an enum type for this instead of a bool, but I don't think go has enums.  Could have two bools instead of one like `ecojiV1 *bool, ecojiV2 *bool` and whenever a rune is seens that is only used by v1 or v2  it will set it to true.  When one is set to true then do not expect to see runes only used be the other.